File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int n = nums.length;
        int res = nums[0] , curr = nums[0];
        for(int i=1; i<n; i++){
            curr = Math.max(nums[i]  , curr+nums[i]);
            res = Math.max(res , curr);
        }
        return res;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int res = nums[0] , sum = nums[0];
        for(int i=1; i<n; i++){
            sum = max(nums[i]  , sum + nums[i]);
            res = max(res , sum);
        }
        return res;
    }
};